### DISCUSSION TITLE
Beginner Q regarding stdout

### DISCUSSION THREAD

**Discussion by dc-acronym** [basilisp-lang/basilisp#1081] [Q&A] [answered]
Apologies for the simple Q, Not sure if this is a Calva related question or a more general one:

When using the basilisp repo, I get fancy rich output to stdout using `rich`.  When doing the same at the live connected calva repl, I don't. Am I missing something about how basilisp renders to stdout? Alternatively, is there a classier, more basilispy way for me to dunk object structure to get a sense of whats going on as I get up to speed? Thanks!

**Comment by ikappaki** [comment, marked as answer]
Hi @dc-acronym,

currently, the nREPL server only redirects the `*out*` stream to the client, meaning it captures output from core functions like `pr`, `print`, etc.

However, `rich` writes directly to `sys/stdout`, which is the terminal or window where the nREPL server was started. So, if you started the server with `basilisp nrepl-server`, the output would appear in that terminal:

```shell
 basilisp nrepl-server
nREPL server started on port 59975 on host 127.0.0.1 - nrepl://127.0.0.1:59975
{'a': 5}
```
Are you primarily interested in displaying Basilisp core structures? If so, you can try the https://github.com/ikappaki/basilisp-pprint pretty printer library, which provides the same interface as the Clojure pretty printer.

Alternatively, if you still want to use `rich`, you can redirect `sys.stdout` to `*out*` (and thus to the client) as follows:

```clojure
(import rich contextlib)

(with [_ (contextlib/redirect_stdout *out*)]
      (rich/print {:a 5})
```

You might want to wrap this into a macro for convenience though.

I hope this helps!

**Reply by dc-acronym** [reply]
This is perfect, thanks!

Basilisp's pprint is 70% of what I was looking for (with the rich redirection covering the 30% of the time I'm too lazy to look up some python function)
